### PR TITLE
Delegate reward ratio should be Type Float64 245

### PR DESCRIPTION
**Pull request by lzxm160**
work on #245 
have tested with api:
```
query{
 hermes(startEpoch: 9550, epochCount:1, 
  rewardAddress: "io12mgttmfa2ffn9uqvn0yn37f4nz43d248l2ga85",
  waiverThreshold: 100) {
  hermesDistribution{
   delegateName
      rewardDistribution{
        voterIotexAddress
        amount
      }
  }
 }
}
query{
  hermes2(startEpoch:10,epochCount:10){
    byDelegate(delegateName:"ducapital"){
      distributionRatio{
        epochNumber
        blockRewardRatio
        epochRewardRatio
        foundationBonusRatio
      }
    }
  }
}
query{
  voting(startEpoch: 1, epochCount: 10){
     rewardSources(voterIotexAddress:
      "io1mtkaz8w8mkzr9l7kg9fls4a4u6v8p4d6wltndx"){
      exist
      delegateDistributions{
        delegateName
    	amount
      }
    }
  }
}
```

### CODE REVIEW

File: queryprotocol/rewards/protocol.go [hunk 1]
```diff
@@ -244,29 +244,14 @@ func (p *Protocol) GetHermesBookkeeping(startEpoch uint64, epochCount uint64, re
...
-			if distributePlan.BlockRewardPercentage < waiverThreshold || distributePlan.EpochRewardPercentage < waiverThreshold ||
-				distributePlan.FoundationBonusPercentage < waiverThreshold {
+			waiverThresholdF := float64(waiverThreshold)
+			if distributePlan.BlockRewardPercentage < waiverThresholdF || distributePlan.EpochRewardPercentage < waiverThresholdF || distributePlan.FoundationBonusPercentage < waiverThresholdF {
 				feeWaiver = false
 			}
-
+			distrReward, err := calculateDistributeReward(distributePlan, rewards)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to calculate reward distribution plan")
```
Comment (lizhefeng) [inline queryprotocol/rewards/protocol.go:253]: failed to calculate distributed reward

File: queryprotocol/rewards/protocol.go [hunk 2]
```diff
@@ -427,6 +386,29 @@ func (p *Protocol) GetRewardSources(startEpoch uint64, epochCount uint64, voterI
 	return delegateDistributions, nil
 }
 
+func calculateDistributeReward(distributePlan *HermesDistributionPlan, rewards *HermesDistributionSource) (*big.Int, error) {
```
Comment (lizhefeng) [inline queryprotocol/rewards/protocol.go:389]: calculatedDistributedReward

File: graphql/resolver.go [hunk 3]
```diff
@@ -1578,12 +1578,12 @@ func (r *queryResolver) getHermes2ByDelegate(ctx context.Context, startEpoch int
...
 			return errors.Wrap(err, "failed to get hermes distribution by delegate name")
 		}
-		for _, ratioInfo := range res {
+		for i, ratioInfo := range res {
 			info := &Ratio{
 				BlockRewardRatio:     ratioInfo.BlockRewardRatio,
 				EpochRewardRatio:     ratioInfo.EpochRewardRatio,
 				FoundationBonusRatio: ratioInfo.FoundationBonusRatio,
-				EpochNumber:          startEpoch,
+				EpochNumber:          startEpoch + i,
```
Comment (lizhefeng) [inline graphql/resolver.go:1585]: Can we use ratioInfo.EpochNumber?

Comment (lzxm160) [reply graphql/resolver.go]: sure

Comment (lzxm160) [reply graphql/resolver.go]: sure

**Pull request merged**